File: identhub/src/main/java/de/solarisbank/identhub/navigation/NaviDirection.java
package de.solarisbank.identhub.navigation;

import android.os.Bundle;

public final class NaviDirection {
    private final int actionId;
    private final Bundle args;

    public NaviDirection(int actionId, Bundle args) {
        this.actionId = actionId;
        this.args = args;
    }

    public NaviDirection(int actionId) {
        this.actionId = actionId;
        this.args = null;
    }

    public int getActionId() {
        return actionId;
    }

    public Bundle getArgs() {
        return args;
    }
}

File: session/src/main/java/de/solarisbank/identhub/session/feature/navigation/NaviDirection.kt
package de.solarisbank.identhub.session.feature.navigation

import android.os.Bundle
import de.solarisbank.identhub.session.feature.navigation.router.COMPLETED_STEP

sealed class NaviDirection {

    class FragmentDirection(
        val actionId: Int,
        val args: Bundle? = null
    ) : NaviDirection()

    class PaymentSuccessfulStepResult(
        val identificationId: String,
        val completedStep: Int = COMPLETED_STEP.VERIFICATION_BANK.index
    ): NaviDirection(), SessionStepResult

    class NextStepStepResult(
        val nextStep: String?,
        val completedStep: Int? = null
    ): NaviDirection(), SessionStepResult

    class ConfirmationSuccessfulStepResult(
        val identificationId: String,
        val completedStep: Int = COMPLETED_STEP.CONTRACT_SIGNING.index
    ): NaviDirection(), SessionStepResult

    class VerificationSuccessfulStepResult(
        val identificationId: String,
        val completedStep: Int
    ): NaviDirection(), SessionStepResult

    class VerificationFailureStepResult(
        val completedStep: Int? = null
    ): NaviDirection(), SessionStepResult
}